### PR TITLE
gpu: miopen: softmax: fix benchdnn failures

### PR DESCRIPTION
**Pull request by t4c1**
# Description

Fixes failures of benchdnn tests of softmax with MIOpen (`./benchdnn --mode=C -v1 --engine=gpu --softmax --batch=test_softmax_gpu`). Two issues are fixed:
 - MIOpen only supports 4-dimensional tensors for softmax. This now throws if higher number of dimensions is provided.
 - Threshold for comparison of half precision floating point results was set wrong.

### CODE REVIEW

File: tests/benchdnn/softmax/softmax.cpp [hunk 1]
```diff
@@ -225,9 +225,10 @@ void skip_invalid_prb(const prb_t *prb, res_t *res) {
 void setup_cmp(compare::compare_t &cmp, const prb_t *prb, data_kind_t kind,
         const args_t &ref_args) {
     const auto trh_dt = (prb->dir & FLAG_FWD) ? prb->ddt : prb->sdt;
-    const bool is_flt_or_dbl = trh_dt == dnnl_f32 || trh_dt == dnnl_f64;
+    const bool is_fp
+            = trh_dt == dnnl_f32 || trh_dt == dnnl_f64 || trh_dt == dnnl_f16;
```
Comment (dzarukin) [inline tests/benchdnn/softmax/softmax.cpp:229]: Please elaborate on "Threshold for comparison of half precision floating point results was set wrong."

The change was done on purpose to catch issues risen due to not storing intermediate results into f32 data type buffers. If proper intermediate storing is not happening for AMD, that must be pointed out and carefully explained.

File: src/gpu/amd/miopen_softmax_impl.hpp [hunk 2]
```diff
@@ -68,8 +70,12 @@ struct miopen_softmax_fwd_impl_t : public miopen_softmax_impl_base_t {
...
         if (pd->has_zero_dim_memory()) return status::success;
 
-        if (pd->ndims() > MIOPEN_DIM_MAX) { return status::invalid_arguments; }
-        ndims = pd->ndims() < 4 ? 4 : pd->ndims();
+        if (pd->ndims() > miopen_softmax_max_ndims) {
+            return status::invalid_arguments;
+        }
+        ndims = pd->ndims() < miopen_softmax_max_ndims
+                ? miopen_softmax_max_ndims
+                : pd->ndims();
```
Comment (dzarukin) [inline src/gpu/amd/miopen_softmax_impl.hpp:78]: Why keeping this logic if the functionality is only designed to work on `ndims = 4`?

The comment related to why `MIOPEN_DIM_MAX` is not working is missing.

File: tests/benchdnn/softmax/softmax.cpp (hunk 1 above)
Comment (t4c1) [reply tests/benchdnn/softmax/softmax.cpp:229]: Fixed.

File: src/gpu/amd/miopen_softmax_impl.hpp (hunk 2 above)
Comment (t4c1) [reply src/gpu/amd/miopen_softmax_impl.hpp]: Removing this logic and adding a comment.

**Comment by t4c1** [comment]
@dzarukin I addressed all your review comments. This is waiting for you to take another look at it.

File: tests/benchdnn/softmax/softmax.cpp (hunk 1 above)
Comment (t4c1) [reply tests/benchdnn/softmax/softmax.cpp]: Sorry when reverting it, I missed this bit. Now it should actually be fixed.

**Pull request merged**